File: backend/app/models/import_job.py
"""ImportJob / ImportRecord 模型 - Epic 25.1."""

from __future__ import annotations

from datetime import datetime

from sqlalchemy import (
    DateTime,
    ForeignKey,
    Index,
    Integer,
    String,
    Text,
    func,
)
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.database import Base


class ImportJob(Base):
    __tablename__ = "import_jobs"
    __table_args__ = (
        Index("ix_import_jobs_operator_created", "operator_id", "created_at"),
    )

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    operator_id: Mapped[int] = mapped_column(
        ForeignKey("users.id"), nullable=False
    )
    job_type: Mapped[str] = mapped_column(String(32), nullable=False)
    status: Mapped[str] = mapped_column(
        String(16), default="pending", nullable=False
    )  # pending | processing | done | failed
    total_count: Mapped[int] = mapped_column(Integer, default=0, nullable=False)
    success_count: Mapped[int] = mapped_column(
        Integer, default=0, nullable=False
    )
    failed_count: Mapped[int] = mapped_column(
        Integer, default=0, nullable=False
    )
    failed_file_path: Mapped[str | None] = mapped_column(
        String(255), nullable=True
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    completed_at: Mapped[datetime | None] = mapped_column(
        DateTime(timezone=True), nullable=True
    )

    records: Mapped[list[ImportRecord]] = relationship(
        back_populates="job", lazy="selectin", cascade="all, delete-orphan"
    )


class ImportRecord(Base):
    __tablename__ = "import_records"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    job_id: Mapped[int] = mapped_column(
        ForeignKey("import_jobs.id", ondelete="CASCADE"), nullable=False
    )
    row_number: Mapped[int] = mapped_column(Integer, nullable=False)
    status: Mapped[str] = mapped_column(String(16), nullable=False)
    error_message: Mapped[str] = mapped_column(
        Text, default="", nullable=False
    )

    job: Mapped[ImportJob] = relationship(back_populates="records")

File: go-backend/internal/model/import_job.go
package model

import "time"

// ImportJob represents a batch user import operation.
type ImportJob struct {
	ID             int64      `json:"id"`
	OperatorID     int64      `json:"operator_id"`
	JobType        string     `json:"job_type"`
	Status         string     `json:"status"` // pending, processing, done, failed
	TotalCount     int        `json:"total_count"`
	SuccessCount   int        `json:"success_count"`
	FailedCount    int        `json:"failed_count"`
	FailedFilePath *string    `json:"failed_file_path"`
	CreatedAt      time.Time  `json:"created_at"`
	CompletedAt    *time.Time `json:"completed_at"`

	Records []ImportRecord `json:"records,omitempty"`
}

// ImportRecord represents a single row result in an import job.
type ImportRecord struct {
	ID           int64  `json:"id"`
	JobID        int64  `json:"job_id"`
	RowNumber    int    `json:"row_number"`
	Status       string `json:"status"`
	ErrorMessage string `json:"error_message"`
}
